lib/doorkeeper/rails/helpers.rb

```diff
@@ -31,10 +31,16 @@ def doorkeeper_render_error
 
       def doorkeeper_render_error_with(error)
         options = doorkeeper_render_options(error) || {}
+        status = if options.delete(:not_found_for_forbidden) &&
+                    error.status == :forbidden
+                   :not_found
+                 else
+                   error.status
+                 end
         if options.blank?
-          head error.status
+          head status
         else
-          options[:status] = error.status
+          options[:status] = status
           options[:layout] = false if options[:layout].nil?
           render options
         end
```

Comment:
I was attempting to avoid leaving the `:not_found_for_forbidden` key in the `options` hash, and that gets awkward when using a helper (it's possible, but then the helper has side effects, and that makes the code harder to read), which is why I wrote it that way. If you'd rather see it split out, and you're OK leaving the key in the `options` hash, I can make the change.